25_exceptions_lab/02.py

```diff
@@ -5,3 +5,17 @@
 Alert the user politely if there was any problem opening the file
 """
 
+import os.path
+import sys
+
+
+fileName = sys.argv[1]
+
+if not (os.path.isfile(fileName)):
```

Comment:
It is generally preferred to perform an action AND catch the exception than to check `os.path.isfile` first:

1. There are tons of reasons why opening a file can fail (no permissions, IO errors, etc.). Verifying something is a file doesn't cover all the cases.

2. Calling `os.path.isfile` is an extra unneeded operation (which is a waste).

3. When `os.path.isfile` fails you still raise an exception, so you'll still need to add an except block to the `with open` command